lib/tracing-limit/src/lib.rs

```diff
@@ -115,23 +117,27 @@ where
         let mut limit_visitor = LimitVisitor::default();
         event.record(&mut limit_visitor);
 
-        let component_name = {
-            let mut scope = ctx
+        let rate_limit_key_values = {
+            let scope = ctx
                 .lookup_current()
                 .into_iter()
                 .flat_map(|span| span.from_root().chain(std::iter::once(span)));
 
-            scope.find_map(|span| {
-                let extensions = span.extensions();
-                extensions
-                    .get::<RateLimitedSpanKeys>()
-                    .and_then(|fields| fields.component_name.clone())
+            scope.fold(vec![], |mut acc, span| {
```

Comment:
For every call to `on_event` we're now allocating a smallish vector, appending to it and then tossing it into a `RateKeyIdentifier`. In situations like these I get pretty curious how [smallvec](https://docs.rs/smallvec/1.6.1/smallvec/) compares to stdlib's vec.

Reply:
Mmm, that's a good call. I'll try it out 👍